core/src/main/scala/org/http4s/ContentCoding.scala

```diff
@@ -18,54 +18,117 @@
   def satisfies(encoding: ContentCoding): Boolean = encoding.satisfiedBy(this)
 
   @deprecated("Use `Accept-Encoding`.isSatisfiedBy(encoding)", "0.16.1")
   def satisfiedBy(encoding: ContentCoding): Boolean =
-    (this.coding.toString == "*" || this.coding == encoding.coding) &&
+    (this === ContentCoding.`*` || this.coding.equalsIgnoreCase(encoding.coding)) &&
       qValue.isAcceptable && encoding.qValue.isAcceptable
 
   def matches(encoding: ContentCoding): Boolean =
-    (this.coding.toString == "*" || this.coding == encoding.coding)
+    (this === ContentCoding.`*` || this.coding.equalsIgnoreCase(encoding.coding))
 
-  override def render(writer: Writer): writer.type = writer << coding << qValue
+  override def equals(o: Any) = o match {
+    case that: ContentCoding =>
+      this.coding.equalsIgnoreCase(that.coding) && this.qValue === that.qValue
+    case _ => false
+  }
 
-  // We want the normal case class generated methods except copy
-  private def copy(coding: CaseInsensitiveString, q: QValue) =
-    ContentCoding(coding, q)
-}
+  private[this] var hash = 0
+  override def hashCode(): Int = {
+    if (hash == 0) {
+      hash = MurmurHash3.mixLast(coding.toLowerCase.##, qValue.##)
+    }
+    hash
+  }
+
+  override def toString = s"ContentCoding(${coding.toLowerCase}, $qValue)"
+
+  override def compare(other: ContentCoding): Int =
+    ContentCoding.order.compare(this, other)
 
-object ContentCoding extends Registry {
-  type Key = CaseInsensitiveString
-  type Value = ContentCoding
+  override def render(writer: Writer): writer.type =
+    ContentCoding.http4sInstancesForContentCoding.render(writer, this)
+}
 
-  implicit def fromKey(k: CaseInsensitiveString): ContentCoding = ContentCoding(k)
+object ContentCoding {
+  def unsafeFromString(coding: String): ContentCoding =
+    fromString(coding).valueOr(throw _)
 
-  implicit def fromValue(v: ContentCoding): CaseInsensitiveString = v.coding
+  def fromString(coding: String): ParseResult[ContentCoding] =
+    parse(coding)
 
-  val `*` : ContentCoding = registerKey("*".ci)
+  val `*` : ContentCoding = new ContentCoding("*")
 
   // http://www.iana.org/assignments/http-parameters/http-parameters.xml#http-parameters-1
-  val compress = registerKey("compress".ci)
-  val deflate = registerKey("deflate".ci)
-  val exi = registerKey("exi".ci)
-  val gzip = registerKey("gzip".ci)
-  val identity = registerKey("identity".ci)
-  val `pack200-gzip` = registerKey("pack200-gzip".ci)
+  val compress = new ContentCoding("compress")
+  val deflate = new ContentCoding("deflate")
+  val exi = new ContentCoding("exi")
+  val gzip = new ContentCoding("gzip")
+  val identity = new ContentCoding("identity")
+  val `pack200-gzip` = new ContentCoding("pack200-gzip")
 
   // Legacy encodings defined by RFC2616 3.5.
-  val `x-compress` = register("x-compress".ci, compress)
-  val `x-gzip` = register("x-gzip".ci, gzip)
+  val `x-compress` = compress
```

Comment:
I'm surprised these are equivalent to their replacements, but I guess they always have been.

Reply:
That's correct